Bastien Durand/week_01/TheWordGuesser/TheWordGuesser.js

```diff
@@ -0,0 +1,84 @@
+// # Homework: The Word Guesser
+//
+// You'll create a simple word guessing game where the user gets infinite tries to guess the word (like Hangman without the hangman, or like Wheel of Fortune without the wheel and fortune).
+//
+// - Create two global arrays: one to hold the letters of the word (e.g. 'F', 'O', 'X'), and one to hold the current guessed letters (e.g. it would start with '\_', '\_', '\_' and end with 'F', 'O', 'X').
+// - Write a function called guessLetter that will:
+// - Take one argument, the guessed letter.
+// - Iterate through the word letters and see if the guessed letter is in there.
+// - If the guessed letter matches a word letter, changed the guessed letters array to reflect that.
+// - When it's done iterating, it should log the current guessed letters ('F__')
+// and congratulate the user if they found a new letter.
+// - It should also figure out if there are any more letters that need to be guessed,
+// and if not, it should congratulate the user for winning the game.
+// - Pretend you don't know the word, and call guessLetter multiple times with various letters to check that your program works.
+
+console.log('To play this game, simply write within the console guessedLetter(" "), Enter your letter within the quotation marks to see a result')
+
+const currentWordLetters = ['M', 'O', 'N', 'K', 'E', 'Y'];
+const currentGuesses = ['_', '_', '_', '_', '_', '_'];
+
+
+const guessedLetter = function (arg) {
+
+
+  if (arg === 'M') {
+    currentGuesses.splice(0, 1, 'M')
+
+  } else if (arg === 'O') {
+    currentGuesses.splice(1, 1, 'O')
+    console.log(`${arg} is Correct`)
+
+  } else if (arg === 'N') {
+    currentGuesses.splice(2, 1, 'N')
+    console.log(`${arg} is Correct`)
+
+  } else if (arg === 'K') {
+    currentGuesses.splice(3, 1, 'K')
+    console.log(`${arg} is Correct`)
+
+  } else if (arg === 'E') {
+    currentGuesses.splice(4, 1, 'E')
+    console.log(`${arg} is Correct`)
+
+  } else if (arg === 'Y') {
+    currentGuesses.splice(5, 1, 'Y')
+    console.log(`${arg} is Correct`)
+
+  } else if (arg !== 'M' || arg !== 'O' || arg !== 'N' || arg !== 'K' || arg !== 'E' || arg !== 'Y' ) {
+    console.log(`${arg} is Incorrect`);
+  }
+
+
+console.log(currentGuesses)
+if (currentGuesses === currentWordLetters) {
+  console.log('YOU WIN')
+}
+
+};
+
+// This doesn't work for some reason :(
+
+if (currentGuesses === currentWordLetters) {
+  console.log('YOU WIN')
+}
+
```

Comment:
Yep you're right! This doesn't work. This is because JavaScript sucks. Kidding. But seriously. The only way to check that two arrays are the same and have the same elements in the same order is to loop through them and check. 
e.g. 
```
var arraysMatch = function (arr1, arr2) {
	// Check if all items exist and are in the same order
	for (var i = 0; i < arr1.length; i ++) {
		if (arr1[i] !== arr2[i]) {
                      return false;
                 }
	}

	// Otherwise, return true
	return true;
};
```

Feel free to give this part of the problem another go and submit again and I'll take a look at it.